compiler/rustc_middle/src/mir/mod.rs

```diff
@@ -117,44 +118,46 @@ impl MirPhase {
     }
 
     /// Parses a `MirPhase` from a pair of strings. Panics if this isn't possible for any reason.
-    pub fn parse(dialect: String, phase: Option<String>) -> Self {
-        match &*dialect.to_ascii_lowercase() {
-            "built" => {
+    pub fn parse(dialect: attrs::MirDialect, phase: Option<attrs::MirPhase>) -> Self {
+        match dialect {
+            attrs::MirDialect::Built => {
+                // Caught during attribute checking.
                 assert!(phase.is_none(), "Cannot specify a phase for `Built` MIR");
                 MirPhase::Built
             }
-            "analysis" => Self::Analysis(AnalysisPhase::parse(phase)),
-            "runtime" => Self::Runtime(RuntimePhase::parse(phase)),
-            _ => bug!("Unknown MIR dialect: '{}'", dialect),
+            attrs::MirDialect::Analysis => Self::Analysis(AnalysisPhase::parse(phase)),
+            attrs::MirDialect::Runtime => Self::Runtime(RuntimePhase::parse(phase)),
         }
     }
 }
 
 impl AnalysisPhase {
-    pub fn parse(phase: Option<String>) -> Self {
+    pub fn parse(phase: Option<attrs::MirPhase>) -> Self {
         let Some(phase) = phase else {
             return Self::Initial;
         };
 
-        match &*phase.to_ascii_lowercase() {
-            "initial" => Self::Initial,
-            "post_cleanup" | "post-cleanup" | "postcleanup" => Self::PostCleanup,
```

Comment:
Sigh, yes. Is this attribute stable in some way or gated behind nightly flags? In the latter case I feel like this should be changed. These strings should at least become symbols if they're not yet